run.py

```diff
@@ -1,6 +1,7 @@
 import openai, time, json, transformers, re, datetime, sys, os
 from mailmerge import MailMerge #pip install docx-mailmerge
 from tqdm import tqdm, trange
+from dotenv import load_dotenv
```

Comment:
load_dotenv is not needed as the os module already used. You can instead use api_key = os.environ.get('OPENAI_KEY')

Reply:
python_dotenv module is needed if you want to use .env files to standardise how environment variables are set across OS's.

Reply:
Nice one!